src/scales/scale.linearbase.js

```diff
@@ -23,16 +25,26 @@ function generateTicks(generationOptions, dataRange) {
 	var precision = generationOptions.precision;
 	var spacing, factor, niceMin, niceMax, numSpaces;
 
+	var rmin = dataRange.min;
+	var rmax = dataRange.max;
+	var isNullOrUndef = helpers.isNullOrUndef;
+
 	// spacing is set to a nice number of the dataRange divided by maxNumSpaces.
 	// stepSize is used as a minimum unit if it is specified.
-	spacing = helpers.niceNum((dataRange.max - dataRange.min) / maxNumSpaces / unit) * unit;
-	numSpaces = Math.ceil(dataRange.max / spacing) - Math.floor(dataRange.min / spacing);
+	spacing = helpers.niceNum((rmax - rmin) / maxNumSpaces / unit) * unit;
+
+	// In case of really small numbers and min / max are undefined, default to rmin / rmax
```

Comment:
This comment seams useless, reading the if condition is clear enough.

Reply:
Perhaps it could be rewritten to explain why the `if` statement is needed. Something like:
```
// Beyond MIN_SPACING floating point numbers being to lose precision such that we can't do the math necessary to generate ticks
```